.mci.yml

```diff
@@ -191,6 +191,48 @@ functions:
                 echo "{ \"releases\": { \"default\": \"$MONGODB_BINARIES\" }}" > $MONGO_ORCHESTRATION_HOME/orchestration.config
                 ./.evergreen/run-orchestration.sh
 
+                # Ensure server on port 27017 is the primary server.
```

Comment:
> by directly connecting to it via the default URI (which does not list the other replset members)

Another alternative to consider: the C driver's `test_framework_get_uri_str_no_auth` constructs a URI with multiple hosts after checking the hello response. Using multiple hosts in the URI may be more robust if tests are expected to step down the primary. AFAICT the C++ driver has no such tests, but there are [specification tests that use replSetStepDown](https://github.com/mongodb/specifications/blob/c4f3fabe9047e5e7a87877d642fa0d885a47351c/source/server-discovery-and-monitoring/tests/unified/rediscover-quickly-after-step-down.yml#L88).

Though setting the priority still seems like an improvement, and I expect should resolve the "not primary" errors.

Reply:
I considered adding the replset hosts to the URIs used by the test framework, but decided against it as many tests use the default-constructed `mongocxx::uri` object to obtain the URI, and updating the test suite to use a test-framework URI constructor like the C Driver seemed too extensive a change for this PR.